rioxarray/_io.py

```diff
@@ -1050,11 +1082,22 @@ def open_rasterio(
             for attr, value in result.attrs.items()
             if not attr.startswith(f"{coord}#")
         }
-    # remove duplicate tags
-    if result.name:
-        result.attrs = {
-            attr: value
-            for attr, value in result.attrs.items()
-            if not attr.startswith(f"{result.name}#")
-        }
-    return result
+    # remove duplicate tags for DataArray
+    if not output_is_dataset: # result.name is not available in a Dataset
+        if result.name:
+            result.attrs = {
+                attr: value
+                for attr, value in result.attrs.items()
+                if not attr.startswith(f"{result.name}#")
+            }
+    # remove duplicate tags for Dataset, otherwise there are issues opening the netCDF file in QGIS
```

Comment:
This is interesting. Mind adding a separate function called `_clean_attrs` and moving the duplicate cleanup there?